### PR TITLE
Wrapping Numerics

### PR DESCRIPTION
**Pull request by nixpulvis**
We want the values on the tape, and the value of the pointer to wrap in our language. Bound checking isn't something brainfuck should do IMO.

### CODE REVIEW

**Comment by nixpulvis** [comment]
Worth noting, the ergonomics of `Wrapping` are not great, but it is nice to see them explicitly.

File: src/lib.rs [hunk 1]
```diff
@@ -76,30 +77,31 @@ impl Instruction {
...
-                interp.ptr = interp.ptr + 1;
+                interp.ptr = interp.ptr + Wrapping(1);
             },
             Instruction::DecPtr => {
-                interp.ptr = interp.ptr - 1;
+                interp.ptr = interp.ptr - Wrapping(1);
             },
             Instruction::IncVal => {
-                interp.tape[interp.ptr] = interp.tape[interp.ptr] + 1;
+                interp.tape[interp.ptr.0] = interp.tape[interp.ptr.0] + Wrapping(1);
```
Comment (offlinemark) [inline src/lib.rs]: What does `.0` do? googling 'rust dot zero' isn't very helpful lol

Comment (nixpulvis) [inline src/lib.rs]: `Wrapping` is a tuple struct so `.0` is the accessor (projection) for the first field of the struct, in this case the wrapped value.

**Comment by offlinemark** [comment]
I get this output when I try to run hello.b

```
mark macbook ~/c/l/r/brainfuck ((85c88d0...)) ❯ cargo run fixtures//hello.b
     Running `target/debug/brainfuck fixtures//hello.b`
hXXX WWW
```

**Comment by nixpulvis** [comment]
In testing these things it's key to play with and without the `--release` flag. Also it'd be great to get a sense of what semantics we really want exactly. I'm pretty happy with having both the values and the pointer be wrapping numbers, but I can also see merit in bound checking the pointer explicitly.

**Comment by nixpulvis** [comment]
Yea this is clearly not a working implementation ATM is it lol. Should add a test for whatever the fuck I broke. I was mostly avoiding writing tests that mock STDIN/STDOUT.

**Comment by nixpulvis** [comment]
Rebased maser with new testing output of programs, and we're catching this bug now.

**Comment by nixpulvis** [comment]
Now we just need to implement something like `Wrapping` for wrapping to the size of the tape. Again, now the more I think about it the more I'm liking checking the pointer at runtime.

**Comment by nixpulvis** [comment]
@mossberg do you have any thoughts on the best course of action for wrapping? I'm thinking about adding an error type for runtime errors and having one returned in these cases.

**Comment by nixpulvis** [comment]
FWIW: http://www.muppetlabs.com/~breadbox/bf/standards.html

**Comment by nixpulvis** [comment]
Merging this as it for now. Might look back at this later in another PR.

**Pull request merged**